test-paper.py

```diff
@@ -0,0 +1,26 @@
+import doctest
+
+files = ["introduction.tex",
+         "architecture.tex",
+         "numerics.tex",
+         "features.tex",
+         "simplification.tex",
+         "calculus.tex",
+         "printers.tex",
+         "solvers.tex",
+         "matrices.tex",
+         "domain_specific.tex"
+        ]
+
+full_text = ""
+
+output_file = "test-paper.txt"
+
+for file in files:
+    with open(file, 'r') as f:
+        full_text += "\n" + f.read()
+
```

Comment:
I suppose the empty lines are to make the doctests work. Perhaps `full_text.replace(r'\end{verbatim}', '')` here would have the same effect.